### PR TITLE
Task#3 As a talent I want to estimate proofs of other talents by givi…

### PR DESCRIPTION
**Pull request by Denis973**
…ng "kudos"

### CODE REVIEW

**Review by LordRenDS — APPROVED** [review]
Nice work

File: src/main/java/com/provedcode/kudos/KudosService.java [hunk 1]
```diff
@@ -1,16 +1,56 @@
...
+
+        Talent talent = userInfo.getTalent();
+
+        TalentProof talentProof = talentProofRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
+                        "Proof with id = %s not found".formatted(id)));
+
+        if (talent.getId() == talentProof.getTalent().getId()) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Talent can’t give “kudos“ to himself");
```
Comment (Maslyna) [inline src/main/java/com/provedcode/kudos/KudosService.java:43]: In this point you should use 409 status code `CONFLICT`

File: src/main/java/com/provedcode/kudos/KudosService.java [hunk 2]
```diff
@@ -1,16 +1,56 @@
...
+        }
+
+        if (kudosRepository.existsByTalent(talent)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Talent can give only one “kudos“ for one proof");
```
Comment (Maslyna) [inline src/main/java/com/provedcode/kudos/KudosService.java]: In this point you should use 409 status code `CONFLICT`

**Review by Maslyna — CHANGES_REQUESTED** [review]
Not ready yet.

**Pull request merged**